Applications/Terminal/Terminal.cpp

```diff
@@ -586,7 +586,10 @@ void Terminal::on_char(byte ch)
         }
         return;
     case '\a':
-        sysbeep();
+        if (m_should_beep)
+            sysbeep();
+        else
+            m_visual_beep_frames = 2;
```

Comment:
What's driving the next 2 repaints after this? Just the cursor blinking or random I/O occurring?
Seems like we could get stuck in "visual beep mode" for quite a long time.

I would suggest something like this instead:
1. Set some kind of `m_beeping_visually` flag.
2. Call `update()` so we repaint soon.
3. Spawn a single-shot CTimer with some short timeout. In the callback, clear the `m_beeping_visually` flag and call `update()` once again.